File: src/Gostio.Services/Database/Entities/AccommodationPhoto.cs
namespace Gostio.Services.Database.Entities;

public class AccommodationPhoto : IListingPhoto
{
    public int Id { get; set; }

    public int AccommodationId { get; set; }

    public Accommodation Accommodation { get; set; } = null!;

    public byte[] Image { get; set; } = null!;

    public string ContentType { get; set; } = null!;

    public bool IsCover { get; set; }

    public int DisplayOrder { get; set; }

    public DateTime UploadedAt { get; set; }
}

File: apps/gostio_desktop/lib/features/accommodations/data/accommodation_photo.dart
import 'package:json_annotation/json_annotation.dart';

part 'accommodation_photo.g.dart';

@JsonSerializable(createToJson: false)
class AccommodationPhoto {
  const AccommodationPhoto({
    required this.id,
    required this.listingId,
    required this.contentType,
    required this.isCover,
    required this.displayOrder,
    required this.sizeInBytes,
    required this.uploadedAt,
  });

  factory AccommodationPhoto.fromJson(Map<String, dynamic> json) =>
      _$AccommodationPhotoFromJson(json);

  final int id;
  final int listingId;
  final String contentType;
  final bool isCover;
  final int displayOrder;
  final int sizeInBytes;
  final DateTime uploadedAt;

  String get contentPath => '/accommodations/$listingId/photos/$id/content';
}
